7_react/react2/week1/Github-user-Search/src/GithubUsersContext.jsx

```diff
@@ -0,0 +1,34 @@
+import React, { createContext, useContext, useState } from 'react';
+
+const GithubUsersContext = createContext();
+
+export function useGithubUsers() {
+    return useContext(GithubUsersContext);
+}
+
+export const GithubUsersProvider = ({ children }) => {
+    const [users, setUsers] = useState([]);
+    const [loading, setLoading] = useState(false);
+    const [error, setError] = useState('');
+
+    const searchUsers = async (query) => {
+        setLoading(true);
+        setError('');
+        try {
+            const response = await axios.get(`https://api.github.com/search/users?q=${query}`);
+            setUsers(response.data.items);
+            if (response.data.items.length === 0) {
+                setError('No results...');
+            }
+        } catch (err) {
+            setError(`Error fetching: ${err.message}`);
+        }
+        setLoading(false);
```

Comment:
Moved setLoading(false) to a finally block to ensure it’s executed regardless of whether the try block succeeds or the catch block is executed.